umbraco/Umbraco-CMS
# Content Security Policy

### Prerequisites


### Description

This PR adds configuration for enabling the 'Content Security Policy' header to umbraco, improving security out of the box. By default the section is commented out. So no changes are in effect immediately, but they are easily configurable.

Testing:

- Change the configuration and verify that the header is added in the dev-tools (see screenshot) and that no errors are present in console. This applies to ALL pages (including generated from the CMS).

![image](https://user-images.githubusercontent.com/28300975/59768080-fb99dc80-92a3-11e9-9c3f-e8a03a702896.png)

- Change the configuration so that content from external sites is not enabled (scriptSrc self="true" hosts="" f.ex.) and verify that when trying to load scripts from external websites they are blocked by CSP. 

Note! 
CSP doesnt work on all browsers the same :) use this for reference
https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP#Browser_compatibility

Unit tests:
Added
